frontend/src/app/login/login.component.ts

```diff
@@ -26,16 +30,17 @@ export class LoginComponent implements OnInit {
     submit(email: string, password: string): void {
         this.auth.login(email, password)
             .subscribe(
-            response => {
+                response => {
+                    this.userEmailService.setEmail(email); // set email for observer form submission
```

Comment:
Not a fan of the comment, since it will be used for more than that single use case. Maybe make the comment broader or remove it.